crates/core/src/time.rs

```diff
@@ -362,3 +375,34 @@ impl From<DurationNanos> for crate::tendermint::Timeout {
         Self::from(std::time::Duration::from(val))
     }
 }
+
+#[cfg(test)]
+mod core_time_tests {
+    use super::*;
+
+    // TODO: if someone wants to take this on, convert this test
```

Comment:
By the way, proptesting these properties might be straightforward. See the code at https://github.com/anoma/namada/blob/879a3268f2edfbdd0da1c98cf0e4bbdac48f5b0d/crates/sdk/src/lib.rs#L1008 .